opendevin/core/config.py

```diff
@@ -91,6 +99,14 @@ def int_value(value, default, config_key):
     ]:
         config[k] = int_value(config[k], v, config_key=k)
 
+    if k in [
+        ConfigType.RUN_AS_DEVIN,
+        ConfigType.USE_HOST_NETWORK,
+        ConfigType.AGENT_MEMORY_ENABLED,
```

Comment:
This fixes an actual bug, memory still got initialized because the check is made for a boolean, and this isn't. 'false' is truthy.

Reply:
@rbren